docs/cloud/azure/architecture.md

```diff
@@ -1 +1,939 @@
+
+HDBs are often expected to be mirrored locally. Some users, (e.g.
+quants) need a subset of the data for heavy analysis and backtesting
+where the performance is critical.
+
+### KX Gateway
+
+In production, a kdb+ system may be accessing multiple timeseries datasets, 
+usually each one representing a different market data source, or
+using the same data, refactored for different schemas. All core
+components of a kdb+tick can handle multiple tables. However, you can
+introduce multiple TPs, RDBs and HDBs based on your fault-tolerance
+requirements. This can result in a large number of kdb+ components and a
+high infrastructure segregation. A KX gateway generally acts as a single
+point of contact for a client. A gateway collects data from the
+underlying services, combines datasets and may perform further data
+operations (e.g. aggregation, joins, pivoting, etc.) before it sends the
+result back to the user.
+
+The specific design of a gateway can vary in several ways according to
+expected use cases. For example, in a hot-hot setup, the gateway can be
+used to query services across availability zones.
+
+The implementation of a gateway is largely determined by the following
+factors.
+
+-   Number of clients or users
+-   Number of services and sites
+-   Requirement for data aggregation
+-   Support of free-form queries
+-   Level of redundancy and failover
+
+The task of the gateway is
+
+-   Check user entitlements and data-access permissions
+-   Provide access to stored procedures, utility functions and business logic
+-   Gain access to data in the required services (TP, RDB, HDB)
+-   Provide the best possible service and query performance
+
+The KX Gateway must be accessible through Azure security rules from all
+clients of the kdb+ service. In addition, the location of the gateway
+service also needs to have visibility to the remaining kdb+ processes
+constituting the full KX service.
+
+## Storage and filesystem
+
+Kdb+ tick architecture needs storage space for three types of data:
+
+*TP log*
+
+:   If the TP needs to handle many updates,
+    then writing to it needs to be fast since slow I/O may delay updates
+    and can even cause data loss. Optionally, you can write updates to
+    the TP log in batches (e.g. every second) as opposed to in real time.
+    You will suffer data loss if TP or instance is halted unexpectedly
+    or stops or restarts, as the recently received updates are not
+    persisted. Nevertheless, you already suffer data loss if a TP
+    process or the Azure VM instance goes down or restarts. The extra
+    second of data loss is probably marginal to the whole outage window.
+
+    If the RDB process goes down, then it can replay data to recover from
+    the TP log. The faster it can recover, the fewer data are waiting in
+    the TP output queue to be processed by the restarted RDB. Hence, a fast
+    read operation is critical for resilience reasons. Using Azure Premium
```

Comment:
Probably worth noting that managed disks are more resilient and would still have the data despite any Azure VM restart or loss.